site-scrapers/NatickMall.js

```diff
@@ -0,0 +1,102 @@
+const sites = require("../data/sites.json");
+const https = require("https");
+
+module.exports = async function GetAvailableAppointments() {
+    console.log("NatickMall starting.");
+    const webData = await ScrapeWebsiteData();
+    console.log("NatickMall done.");
+    return {
+        ...sites.NatickMall,
+        ...webData,
+    };
+};
+
+async function ScrapeWebsiteData() {
+    const tokenUrl =
+        "https://home.color.com/api/v1/get_onsite_claim?partner=natickmall";
+    // Get a request token
+    const tokenPromise = new Promise((resolve) => {
```

Comment:
You don't actually need to re-generate a token each time. You can just hardcode any claim token you get and use it to query for availability forever. They never expire.

Reply:
How would that be provable? How do we know it and know those unwritten guarantees might not go away?

Note that this code has been merged (Feb. 19) and is in production now.